delta-lake/src/main/java/org/apache/iceberg/delta/BaseSnapshotDeltaLakeTableAction.java

```diff
@@ -0,0 +1,396 @@
+      // DeleteFiles case
+      DeleteFiles deleteFiles = transaction.newDelete();
+      filesToRemove.forEach(deleteFiles::deleteFile);
+      deleteFiles.commit();
+    }
+  }
+
+  private DataFile buildDataFileFromAction(Action action, Table table) {
+    PartitionSpec spec = table.spec();
+    String path;
+    long fileSize;
+    Long nullableFileSize;
+    Map<String, String> partitionValues;
+
+    if (action instanceof AddFile) {
+      AddFile addFile = (AddFile) action;
+      path = addFile.getPath();
+      nullableFileSize = addFile.getSize();
+      partitionValues = addFile.getPartitionValues();
+    } else if (action instanceof RemoveFile) {
+      RemoveFile removeFile = (RemoveFile) action;
+      path = removeFile.getPath();
+      nullableFileSize = removeFile.getSize().orElse(null);
+      partitionValues = removeFile.getPartitionValues();
+    } else {
+      throw new ValidationException(
+          "Unexpected action type for Delta Lake: %s", action.getClass().getSimpleName());
+    }
+
+    String fullFilePath = getFullFilePath(path, deltaLog.getPath().toString());
+    // For unpartitioned table, the partitionValues should be an empty map rather than null
+    Preconditions.checkArgument(
+        partitionValues != null,
+        String.format("File %s does not specify a partitionValues", fullFilePath));
+
+    FileFormat format = determineFileFormatFromPath(fullFilePath);
+    InputFile file = deltaLakeFileIO.newInputFile(fullFilePath);
+
+    // If the file size is not specified, the size should be read from the file
+    if (nullableFileSize != null) {
+      fileSize = nullableFileSize;
+    } else {
+      fileSize = file.getLength();
+    }
+
+    // get metrics from the file
+    MetricsConfig metricsConfig = MetricsConfig.forTable(table);
+    String nameMappingString = table.properties().get(TableProperties.DEFAULT_NAME_MAPPING);
+    NameMapping nameMapping =
+        nameMappingString != null ? NameMappingParser.fromJson(nameMappingString) : null;
+    Metrics metrics = getMetricsForFile(file, format, metricsConfig, nameMapping);
+
+    String partition =
+        spec.fields().stream()
+            .map(PartitionField::name)
+            .map(name -> String.format("%s=%s", name, partitionValues.get(name)))
+            .collect(Collectors.joining("/"));
+
+    return DataFiles.builder(spec)
+        .withPath(fullFilePath)
+        .withFormat(format)
+        .withFileSizeInBytes(fileSize)
+        .withMetrics(metrics)
+        .withPartitionPath(partition)
+        .build();
+  }
+
+  private FileFormat determineFileFormatFromPath(String path) {
+    if (path.endsWith(PARQUET_SUFFIX)) {
+      return FileFormat.PARQUET;
+    } else if (path.endsWith(AVRO_SUFFIX)) {
+      return FileFormat.AVRO;
+    } else if (path.endsWith(ORC_SUFFIX)) {
+      return FileFormat.ORC;
+    } else {
+      throw new ValidationException("Cannot determine file format from path %s", path);
```

Comment:
nit: not sure if we continue to handle non-parquet suffix and consider this as "cannot determine" rather than "do not support" here after we removed dependencies, as now they will probably fail with `Cannot get metrics from file format`

Reply:
Thank you for pointing this out. I removed support for orc and avro and changed the error message to "do not support..."